### PR TITLE
WAT-480 As engineer I want integration newWindow() function to open Tab into TestRing

### PR DESCRIPTION
**Pull request by danbao**
### Background
 We had a case need to open new Tab

### Problem
I found There haven't integration [newWindow()](http://webdriver.io/api/window/newWindow.html) into TestRing v0.4.45 [plugin-selenium-driver](https://github.com/ringcentral/testring/blob/v0.4.45/packages/plugin-selenium-driver/src/plugin/index.ts)

### CODE REVIEW

File: packages/plugin-selenium-driver/src/plugin/index.ts [hunk 1]
```diff
@@ -288,6 +288,15 @@ export class SeleniumPlugin implements IBrowserProxyPlugin {
         }
     }
 
+    public async newWindow(applicant: string, val: string, windowName: string, windowFeatures: string) {
+        await this.createClient(applicant);
+        const client = this.getBrowserClient(applicant);
+
+        if (client) {
+            return this.wrapWithPromise(client.newWindow(val, windowName, windowFeatures));
```
Comment (johnthecat) [inline packages/plugin-selenium-driver/src/plugin/index.ts]: In closest feature we will have puppeteer plugin too, so we can't use `windowFeatures`, it's too specific.
You can pass it like an object 
```
{
    width: 1000,
    height: 500,
}
```
and convert to string inside this function.

File: packages/plugin-selenium-driver/src/plugin/index.ts [hunk 2]
```diff
@@ -288,6 +288,15 @@ export class SeleniumPlugin implements IBrowserProxyPlugin {
         }
     }
 
+    public async newWindow(applicant: string, val: string, windowName: string, windowFeatures: string) {
```
Comment (johnthecat) [inline packages/plugin-selenium-driver/src/plugin/index.ts]: `openWindow`, maybe? We don't use exactly webdriver.io API, so it's better to use more friendly names.

Comment (danbao) [reply packages/plugin-selenium-driver/src/plugin/index.ts]: Ok, I will rename to openWindow

File: packages/plugin-selenium-driver/src/plugin/index.ts (hunk 1 above)
Comment (danbao) [reply packages/plugin-selenium-driver/src/plugin/index.ts]: Sorry, I don't catch you means. could you please help refactor this part of code?

**Comment by johnthecat** [comment]
Sorry, comment about renaming `newWindow` to `openWindow` was incorrect. After discussion with @flops we decided to keep `newWindow` name, because there is no any shared API concept yet

File: packages/plugin-selenium-driver/src/plugin/index.ts (hunk 1 above)
Comment (johnthecat) [reply packages/plugin-selenium-driver/src/plugin/index.ts]: I think argument `windowFeatures` should be an object, not string.
You can transform it to string inside this function later.

Comment (danbao) [reply packages/plugin-selenium-driver/src/plugin/index.ts]: done

File: packages/plugin-selenium-driver/src/plugin/index.ts [hunk 3]
```diff
@@ -288,6 +288,16 @@ export class SeleniumPlugin implements IBrowserProxyPlugin {
         }
     }
 
+    public async newWindow(applicant: string, val: string, windowName: string, windowFeatures: object) {
+        await this.createClient(applicant);
+        const client = this.getBrowserClient(applicant);
+        let args = JSON.stringify(windowFeatures);
```
Comment (flops) [inline packages/plugin-selenium-driver/src/plugin/index.ts]: If the documentation is right, the args string should be like this:
`'width=420,height=230,resizable,scrollbars=yes,status=1`
JSON.stringify is not correct one to use.

File: packages/types/src/browser-proxy/index.ts [hunk 4]
```diff
@@ -31,6 +31,8 @@ export interface IBrowserProxyPlugin {
 
     url(applicant: string, val: string): Promise<any>;
 
+    newWindow(applicant: string, val: string, windowName: string, windowFeatures: object): Promise<any>;
```
Comment (flops) [inline packages/types/src/browser-proxy/index.ts]: https://developer.mozilla.org/en-US/docs/Web/API/Window/open#Window_features - here is a window features list to define in object.

File: packages/plugin-selenium-driver/src/plugin/index.ts (hunk 3 above)
Comment (danbao) [reply packages/plugin-selenium-driver/src/plugin/index.ts]: done

File: packages/plugin-selenium-driver/src/plugin/index.ts [hunk 5]
```diff
@@ -288,6 +294,16 @@ export class SeleniumPlugin implements IBrowserProxyPlugin {
         }
     }
 
+    public async newWindow(applicant: string, val: string, windowName: string, windowFeatures: object) {
+        await this.createClient(applicant);
+        const client = this.getBrowserClient(applicant);
+        let args = stringifyWindowFeatures(windowFeatures);
```
[the remark this answers is not in the document]
Comment (danbao) [reply packages/plugin-selenium-driver/src/plugin/index.ts]: OK

**Pull request merged**